### PR TITLE
GEOMESA-2316 Kafka - Support geometries with extents

### PR DESCRIPTION
**Pull request by elahrvivaz**
* SizeSeparatedBucketIndex uses a size-separated index to store non-points

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
note: stacks on top of https://github.com/locationtech/geomesa/pull/1992

File: geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaDataStore.scala [hunk 1]
```diff
@@ -83,7 +83,10 @@ class KafkaDataStore(val config: KafkaDataStoreConfig)
           } else {
             config.indices.cqAttributes
           }
+          // note: handles points vs non-points internally
           GeoCQIndexSupport(sft, attributes, config.indices.resolutionX, config.indices.resolutionY)
+        } else if (sft.nonPoints) {
+          SizeSeparatedBucketIndexSupport(sft, config.indices.resolutionX / 360d, config.indices.resolutionY / 180d)
```
Comment (jnh5y) [inline geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaDataStore.scala]: Looks like the SizeSeparatedBucket is not compatible with the CQEngine support?

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/SizeSeparatedBucketIndex.scala [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      }
+
+      override def next(): T = iter.next()
+    }
+
+  }
+}
+
+object SizeSeparatedBucketIndex {
+  // TODO these are somewhat arbitrary
```
Comment (jnh5y) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/SizeSeparatedBucketIndex.scala]: Update to point to an actual ticket?

Also, I had good clean fun with the SSI only realizing some of the 'sizes', so I do like this approach.

File: geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaDataStore.scala (hunk 1 above)
Comment (elahrvivaz) [reply geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaDataStore.scala]: it is, but it gets handled by cqEngine (see note two lines up)

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/SizeSeparatedBucketIndex.scala (hunk 2 above)
Comment (jnh5y) [reply geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/SizeSeparatedBucketIndex.scala]: Configurations here can be a follow-on ticket.

**Comment by jnh5y** [comment]
Add a ticket for anti-meridian handling?

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/SizeSeparatedBucketIndex.scala (hunk 2 above)
Comment (elahrvivaz) [reply geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/SizeSeparatedBucketIndex.scala]: added a ticket

**Comment by elahrvivaz** [comment]
added https://geomesa.atlassian.net/browse/GEOMESA-2323 for AM handling

**Pull request merged**